src/RegistrationManager.ts

```diff
@@ -1,549 +1,569 @@
+    if (bigManagerBalance.gte(txCost)) {
+      log.info(
+        `Sending manager RBTC balance ${bigManagerBalance.toString()} to owner`
+      );
+      const details: SendTransactionDetails = {
+        signer: this._managerAddress,
+        serverAction: ServerAction.VALUE_TRANSFER,
+        destination: this._ownerAddress ?? '',
+        gasLimit: BigNumber.from(bigGasLimit.toString()),
+        gasPrice,
+        value: BigNumber.from(bigValue.toString()),
+        creationBlockNumber: currentBlock,
+      };
+      const { txHash } = await this._transactionManager.sendTransaction(
+        details
+      );
+      transactionHashes.push(txHash);
+    } else {
+      log.error(
+        `manager balance too low: ${bigManagerBalance.toString()}, tx cost: ${txCost.toString()}`
+      );
     }
 
-    async isRegistered(): Promise<boolean> {
-        const isRegistrationCorrect = this._isRegistrationCorrect();
-        return (
-            this.stakeRequired.isSatisfied &&
-            this.isStakeLocked &&
-            isRegistrationCorrect
-        );
+    return transactionHashes;
+  }
+
+  async _sendWorkersEthBalancesToOwner(
+    currentBlock: number
+  ): Promise<string[]> {
+    // sending workers' balance to owner (currently one worker, todo: extend to multiple)
+    const transactionHashes: string[] = [];
+    const gasPrice = await this._contractInteractor.provider.getGasPrice();
+    const bigGasPrice = BigNumberJs(gasPrice.toString());
+    const gasLimit = BigNumber.from(mintxgascost);
+    const txCost = bigGasPrice.multipliedBy(gasLimit.toString());
+
+    const workerBalance = await this._contractInteractor.getBalance(
+      this._workerAddress
+    );
+    const bigWorkerBalance = new BigNumberJs(workerBalance.toString());
+
+    const bigValue = bigWorkerBalance.minus(txCost);
+    if (bigWorkerBalance.gte(txCost)) {
+      log.info(
+        `Sending workers' RBTC balance ${bigWorkerBalance.toString()} to owner`
+      );
+      const details: SendTransactionDetails = {
+        signer: this._workerAddress,
+        serverAction: ServerAction.VALUE_TRANSFER,
+        destination: this._ownerAddress ?? '',
+        gasLimit,
+        gasPrice: BigNumber.from(bigGasPrice.toString()),
+        value: BigNumber.from(bigValue.toString()),
+        creationBlockNumber: currentBlock,
+      };
+      const { txHash } = await this._transactionManager.sendTransaction(
+        details
+      );
+      transactionHashes.push(txHash);
+    } else {
+      log.info(
+        `balance too low: ${bigWorkerBalance.toString()}, tx cost: ${txCost.toString()}`
+      );
     }
 
-    printNotRegisteredMessage(): void {
-        if (this._isRegistrationCorrect()) {
-            return;
-        }
-        const message = `\nNot registered yet. Prerequisites:
-${this.balanceRequired.description}
-${this.stakeRequired.description}
+    return transactionHashes;
+  }
+
+  // FIXME the relayManager needs to be send as parameter
```

Comment:
Its a blocker, it works but it does not work correctly if we have more than 1 server. Since we are removing the ContractInteractor it would require some major changes.

Reply:
updated